File: 1.cpp
#include <bits/stdc++.h>
int height = 20;
int width = 20;
bool **grid;
int x_width = 30;
int y_width = 30;
void set_path(int x, int y){
    grid[y][x] = false;
}

void set_wall(int x, int y){
    grid[y][x] = true;
}

bool is_wall(int x,int y){
    if(x>=0 && x < width && y >= 0 && y < height){
        return grid[y][x];
    }
    else{
        return false;
    }
}

void create_maze(int x,int y){
    set_path(x,y);
    // int all_directions[4][2] = {{1, 0}, {-1, 0}, {0, 1}, {0, -1}};
    std::vector<std::tuple<int,int>> all_directions;
    all_directions.push_back(std::tuple<int,int>( 1 , 0));
    all_directions.push_back(std::tuple<int,int>(-1 , 0));
    all_directions.push_back(std::tuple<int,int>( 0 , 1));
    all_directions.push_back(std::tuple<int,int>( 0 ,-1));
    std::random_shuffle(all_directions.begin(),all_directions.end());
    while(all_directions.size() > 0){
        auto direction_to_try = all_directions.back();
        all_directions.pop_back();
        int node_x = x + (std::get<0>(direction_to_try))*2;
        int node_y = y + (std::get<1>(direction_to_try))*2;

        if(is_wall(node_x,node_y)){
            int link_cell_x = x + (std::get<0>(direction_to_try));
            int link_cell_y = y + (std::get<1>(direction_to_try));
            set_path(link_cell_x,link_cell_y);
            create_maze(node_x,node_y);
        }
    }
    return;
}

std::string printing(){
    std::string string = "";
    std::map<bool,std::string> conv = {
        {true,"█"},
        {false," "}
        };
    for (int y=0;y<height;y++){
        for (int x=0;x<width;x++){
            string += conv[grid[y][x]];
        }
        string += "\n";
    }
    return string;
}

int main(){
    srand(time(0));
    width = int(width/2)*2 + 1;
    height = int(height/2)*2 + 1;
    grid = new bool*[height];
    for(int y=0;y<height;y++){
        grid[y] = new bool[width];
        for(int x=0;x<width;x++){
            grid[y][x] = true;
        }
    }
    std::tuple<int,int> **grid1;
    grid1 = new std::tuple<int,int>*[height];
    for(int y=0;y<height;y++){
        grid1[y] = new std::tuple<int,int>[width];
        for(int x=0;x<width;x++){
            grid1[y][x] = std::tuple<int,int>(y*30+5,-300 + x*30);
        }
    }
    create_maze(10,10);
    for(int y=0;y<height;y++){
        for(int x=0;x<width;x++){
            std::cout << grid[y][x];
        }
        std::cout << "\n";
    }
    for(int y=0;y<height;y++){
        for(int x=0;x<width;x++){
            std::cout << "("<< std::get<0>(grid1[y][x])<<","<<std::get<1>(grid1[y][x])<<"),\t";
        }
        std::cout << "\n";
    }
    int j=0;
    std::vector<std::tuple<int,int,int,int,int>> grid2;
    // insert box co-ordinates :) you can use push front is push back not used :)
    grid2.push_back(std::tuple<int,int,int,int,int>(-300,360,0,0,1));
    grid2.push_back(std::tuple<int,int,int,int,int>( 330,360,0,0,1));
    grid2.push_back(std::tuple<int,int,int,int,int>( 330,290.01,0,0,1));
    grid2.push_back(std::tuple<int,int,int,int,int>(-300,360,0,0,1));
    grid2.push_back(std::tuple<int,int,int,int,int>(-300,290.01,0,0,1));
    grid2.push_back(std::tuple<int,int,int,int,int>( 330,290.01,0,0,1));
    j+=6;
    for(int y=0;y<height;y++){
        for(int x=0;x<width;x++){
            // std::cout<<"timerun : "<<grid[y][x]<< " " <<std::endl;
            if(!grid[y][x]){
                grid2.push_back(std::tuple<int,int,int,int,int>(-300 + x*x_width,y*y_width-340,1,1,1));  // (-,+)
                grid2.push_back(std::tuple<int,int,int,int,int>(-300 + (x+1)*x_width,y*y_width-340,1,1,1));  // (+,+)
                grid2.push_back(std::tuple<int,int,int,int,int>(-300 + (x+1)*x_width,(y+1)*y_width-340,1,1,1));  // (+,-)
                grid2.push_back(std::tuple<int,int,int,int,int>(-300 + x*x_width,y*y_width-340,1,1,1));  // (-,+)
                grid2.push_back(std::tuple<int,int,int,int,int>(-300 + x*x_width,(y+1)*y_width-340,1,1,1));  // (-,-)
                grid2.push_back(std::tuple<int,int,int,int,int>(-300 + (x+1)*x_width,(y+1)*y_width-340,1,1,1));  // (+,-)
                j+=6;
            }
            else{
                grid2.push_back(std::tuple<int,int,int,int,int>(-300 + x*x_width,y*y_width-340,0,0,0));  // (-,+)
                grid2.push_back(std::tuple<int,int,int,int,int>(-300 + (x+1)*x_width,y*y_width-340,0,0,0));  // (+,+)
                grid2.push_back(std::tuple<int,int,int,int,int>(-300 + (x+1)*x_width,(y+1)*y_width-340,0,0,0));  // (+,-)
                grid2.push_back(std::tuple<int,int,int,int,int>(-300 + x*x_width,y*y_width-340,0,0,0));  // (-,+)
                grid2.push_back(std::tuple<int,int,int,int,int>(-300 + x*x_width,(y+1)*y_width-340,0,0,0));  // (-,-)
                grid2.push_back(std::tuple<int,int,int,int,int>(-300 + (x+1)*x_width,(y+1)*y_width-340,0,0,0));  // (+,-)
                j+=6;
            }
        }
    }
    std::cout << "Here are vertices : \n";
    int lamda = 0;
    for(auto k = grid2.begin();k!=grid2.end();k++){
        // if(lamda%6==0){
        //     lamda = 0;
        //     std::cout <<"\n"; 
        // }
        std::cout <<std::get<0>(*k)<<".0,"<<std::get<1>(*k)<<".0,"<<std::get<2>(*k)<<".0,"<<std::get<3>(*k)<<".0,"<<std::get<4>(*k)<<".0,\n";
        j++;
        // lamda++;
    }
    std::cout <<"\n\n"<< grid2.size() << " : " << j << std::endl;
    std::cout <<"\n\n";
    std::cout << printing();
}

File: 1.py
import random

class Maze:
    def __init__(self, width, height):
        self.width = width // 2 * 2 + 1
        self.height = height // 2 * 2 + 1

        # this creates a 2d-array for your maze data (False: path, True: wall)
        self.cells = [
                      [True for x in range(self.width)] 
                      for y in range(self.height)
                     ]

    def set_path(self, x, y):
        self.cells[y][x] = False

    def set_wall(self, x, y):
        self.cells[y][x] = True

    # a function to return if the current cell is a wall,
    #  and if the cell is within the maze bounds
    def is_wall(self, x, y):
        # checks if the coordinates are within the maze grid
        if 0 <= x < self.width and 0 <= y < self.height:
            # if they are, then we can check if the cell is a wall
            return self.cells[y][x]
        # if the coordinates are not within the maze bounds, we don't want to go there
        else:
            return False

    def create_maze(self, x, y):
        # set the current cell to a path, so that we don't return here later
        self.set_path(x, y)
        # we create a list of directions (in a random order) we can try
        all_directions = [[1, 0], [-1, 0], [0, 1], [0, -1]]
        random.shuffle(all_directions)

        # we keep trying the next direction in the list, until we have no directions left
        while len(all_directions) > 0:

            # we remove and return the last item in our directions list
            direction_to_try = all_directions.pop()

            # calculate the new node's coordinates using our random direction.
            # we *2 as we are moving two cells in each direction to the next node
            node_x = x + (direction_to_try[0] * 2)
            node_y = y + (direction_to_try[1] * 2)

            # check if the test node is a wall (eg it hasn't been visited)
            if self.is_wall(node_x, node_y):
                # success code: we have found a path

                # set our linking cell (between the two nodes we're moving from/to) to a path
                link_cell_x = x + direction_to_try[0]
                link_cell_y = y + direction_to_try[1]
                self.set_path(link_cell_x, link_cell_y)

                # "move" to our new node. remember we are calling the function every
                #  time we move, so we call it again but with the updated x and y coordinates
                self.create_maze(node_x, node_y)
        return

    def __str__(self):
        string = ""
        conv = {
            True: "█",
            False: "  "
        }
        for y in range(self.height):
            for x in range(self.width):
                string += conv[self.cells[y][x]]
            string += "\n"
        return string

maze = Maze(10,10)
maze.create_maze(10,10)
print(maze)